src/QueryM.purs

```diff
@@ -179,9 +182,28 @@ type QueryConfig =
   , usedTxOuts :: UsedTxOuts
   , networkId :: NetworkId
   , slotConfig :: SlotConfig
+  | r
   }
 
-type QueryM (a :: Type) = ReaderT QueryConfig Aff a
+type DefaultQueryConfig = QueryConfig ()
+
+type QueryM (a :: Type) = ReaderT DefaultQueryConfig Aff a
+
+type QueryMExtended (r :: Row Type) (a :: Type) = ReaderT (QueryConfig r) Aff a
+
+liftQueryM :: forall (r :: Row Type) (a :: Type). QueryM a -> QueryMExtended r a
+liftQueryM = withReaderT toDefaultQueryConfig
+  where
+  toDefaultQueryConfig :: forall (r :: Row Type). QueryConfig r -> DefaultQueryConfig
```

Comment:
Let's write this as `toDefaultQueryConfig :: QueryConfig r -> DefaultQueryConfig` to prevent shadowing, or use a different letter to `r`.